src/base/projects/View.svelte

```diff
@@ -24,42 +24,45 @@
   import Message from "@app/Message.svelte";
 
   export let wallet: Wallet;
-  export let urn: string;
-  export let seed: string | null;
-  export let profile: string | null;
-  export let peer: string | null;
+  export let activeRoute: ProjectRoute;
 
-  // Casting activeRouteStore to a projectRoute store
-  $: activeRoute = activeRouteStore as Writable<ProjectRoute>;
+  $: urn = activeRoute.params.urn;
+  $: seed = activeRoute.params.seed ?? null;
+  $: profile = activeRoute.params.profile ?? null;
+  $: peer = activeRoute.params.peer ?? null;
 
-  $: searchParams = new URLSearchParams($activeRoute.params.search || "");
+  $: searchParams = new URLSearchParams(activeRoute.params.search || "");
   $: issueFilter = (searchParams.get("state") as IssueState) || "open";
   $: patchFilter = (searchParams.get("state") as PatchState) || "proposed";
 
   // Passing peer as param to allow to react to peer change.
   const getProject = async (peer: string | null) => {
     const project = await proj.Project.get(urn, peer, profile, seed, wallet);
-    if ($activeRoute.params.route) {
+    if (activeRoute.params.route) {
       const { revision, path } = proj.parseRoute(
-        $activeRoute.params.route,
+        activeRoute.params.route,
         project.branches,
       );
       // Updating revision and patch with resolved route and nulling the route param.
-      const params = {
-        ...$activeRoute.params,
-        revision,
-        path,
-        route: null,
+      const params: ProjectRoute = {
+        type: "projects",
+        params: {
+          ...activeRoute.params,
+          revision,
+          path,
+          route: null,
+        },
       };
-      activeRoute.update(s => ({ ...s, params }));
+      router.activeRouteStore.set(params);
     }
 
     return project;
   };
 
   // Content can be altered in child components.
-  $: revision = $activeRoute.params.revision || null;
-  $: content = $activeRoute.params.content || "tree";
+  $: revision = activeRoute.params.revision || null;
+
+  $: console.log(activeRoute.params.activeView);
```

Comment:
This can be eventually removed :eyes: 😅

Reply:
62b004222c52666f9e13b3c346bda5d8fa980cf3